Marlin/Marlin_main.cpp

```diff
@@ -921,6 +926,17 @@ void gcode_line_error(const char* err, bool doFlush = true) {
   serial_count = 0;
 }
 
+void clear_buffer() {
+  for (uint8_t i=0; i < BUFSIZE; i++)
+    for (uint8_t j=0; j < MAX_CMD_SIZE; j++)
+      command_queue[i][j] = NULL;
```

Comment:
If the first character in a string is nul, the string is considered empty. So there's no need to clear the entire string.